client/ws.ts

```diff
@@ -59,50 +79,70 @@ export class Remote {
         }
         const parsedData = JSON.parse(payloadData);
 
-        //Processes for the method 'emitBatch':
-        if (Array.isArray(parsedData) && !isOnetime && parsedData.length > 0) {
-          const invalid = parsedData.map(validateResponse).find((res) =>
+        // Process for the method 'emitBatch':
+        if (Array.isArray(parsedData) && parsedData.length > 0) {
+          const rpcResponses = parsedData.map(validateResponse);
+          const invalid = rpcResponses.find((res) =>
             !isObject(res.result) || res.result.event !== "emitted"
           );
           if (invalid) {
             throw new BadServerDataError(
-              invalid.id ? invalid.id : null,
+              invalid.id || null,
               "The server returned an invalid batch response.",
               -32004,
             );
           } else {
-            continue;
+            for (const res of rpcResponses) {
+              if ((res.result as JsonObject).id === rpcRequest.id) {
+                yield res.result;
+              }
+            }
           }
         } else {
           const rpcResponse = validateResponse(parsedData);
           if (
-            !isOnetime &&
             isObject(rpcResponse.result) &&
             rpcResponse.result.id === rpcRequest.id
           ) {
-            if (
-              rpcResponse.result.event === "subscribed" ||
-              rpcResponse.result.event === "emitted"
-            ) {
-              continue;
-            }
-            if (rpcResponse.result.event === "unsubscribed") {
-              break;
-            }
-          }
-          if (rpcResponse.id === rpcRequest.id) {
-            yield rpcResponse.result;
-            if (isOnetime) {
-              break;
+            switch (rpcResponse.result.event) {
+              case "subscribed":
+                continue;
+              case "unsubscribed":
+                break;
+              case "emitted":
+                yield rpcResponse.result;
+                break;
```

Comment:
Same here. We want to let the subscriber know about a subscription event, right?